EuG SMEs feedback (#1437)
codenvy/che-docs: src/main/pages/che-7/end-user-guide/con_che-theia-task-types.adoc

[id="che-theia-task-types_{context}"]
= Che-Theia task types

The `commands` section in a devfile allows a user to automate certain actions when running a Che workspace.

There are three types of Che commands:

> * `exec`: Each command of type `exec` is translated into a Theia task of type `che`. A `che` command executes a shell command, but in contrast to regular Theia shell-type tasks, a user can choose a workspace container for the command execution.

Reviewer: ```suggestion
* `exec`: Each command of type `exec` is translated into a Che-Theia task of type `che`. A `che` command executes a shell command, but in contrast to regular Che-Theia shell-type tasks, a user can choose a workspace container for the command execution.
```

* `vscode-task`: `vscode-task` is a single command type that copies the contents of the `referenceContentfield` of the specific command directly to the user-level task configurations in Che-Theia.
* `vscode-launch`: Functions like a `vscode-task` command, but instead of being copied into the task configurations, the content of the `referenceContent` field is copied into the `launch.json` configuration file.

Tasks of type `che`, also known as `exec` commands, can be executed from the *Terminal->Run Task* menu or by clicking them in the *My Workspace* panel.
Other tasks are only available via *Terminal->Run Task*. Launch configurations are available in the Che-Theia debugger.

.Example:

The following example contains a  `che` task and `vscode-task` task sample.

> The configuration in link:https://www.eclipse.org/che/docs/che-7/defining-custom-commands-for-che-theia/[Defining custom commands for Che-Theia] sets up the `theia:watch` and `theia:build` tasks in the `theia workspace:

Reviewer: ```suggestion
The configuration in link:https://www.eclipse.org/che/docs/che-7/defining-custom-commands-for-che-theia/[Defining custom commands for Che-Theia] sets up the `theia:watch` and `theia:build` tasks in the `theia` workspace:
```

> * `theia:watch` that will run the theia project in `watch` mode.

Reviewer: ```suggestion
* `theia:watch`, which runs the `theia` project in `watch` mode.
```

> * `theia:build` that will build the `theia` project in the `che-dev` component.

Reviewer: ```suggestion
* `theia:build`, which builds the `theia` project in the `che-dev` component.
```
